src/interfaces/pyo3/hybrid_gpsw_aes.rs

```diff
@@ -0,0 +1,254 @@
+    // Parse user decryption key
+    let user_decryption_key = UserDecryptionKey::from_bytes(&user_decryption_key_bytes)?;
+
+    //
+    // Finally decrypt symmetric key using given user decryption key
+    let cleartext_header: ClearTextHeader<Aes256GcmCrypto> =
+        core_decrypt_hybrid_header::<Gpsw<Bls12_381>, Aes256GcmCrypto>(
+            &user_decryption_key,
+            &encrypted_header_bytes,
+        )
+        .map_err(|e| PyTypeError::new_err(format!("Error decrypting header: {e}")))?;
+
+    let metadata = cleartext_header
+        .meta_data
+        .to_bytes()
+        .map_err(|e| PyTypeError::new_err(format!("Serialize metadata failed: {e}")))?;
+
+    Ok((cleartext_header.symmetric_key.to_bytes(), metadata))
+}
+
+/// Symmetrically Encrypt plaintext data in a block.
+#[pyfunction]
+pub fn encrypt_hybrid_block(
+    symmetric_key_bytes: Vec<u8>,
+    uid_bytes: Option<Vec<u8>>,
+    block_number: Option<usize>,
+    plaintext_bytes: Vec<u8>,
+) -> PyResult<Vec<u8>> {
+    //
+    // Check `plaintext_bytes` input param
+    if plaintext_bytes.is_empty() {
+        return Err(PyTypeError::new_err("Plaintext value is empty"));
+    }
+
+    //
+    // Parse symmetric key
+    let symmetric_key =
+        <Aes256GcmCrypto as SymmetricCrypto>::Key::try_from_bytes(symmetric_key_bytes)
+            .map_err(|e| PyTypeError::new_err(format!("Deserialize symmetric key failed: {e}")))?;
+
+    //
+    // Parse other input params
+    let uid = uid_bytes.map_or_else(Vec::new, |v| v.to_vec());
+    let block_number_value = block_number.unwrap_or(0);
+
+    //
+    // Encrypt block
+    let ciphertext =
+        core_encrypt_hybrid_block::<Gpsw<Bls12_381>, Aes256GcmCrypto, MAX_CLEAR_TEXT_SIZE>(
+            &symmetric_key,
+            &uid,
+            block_number_value as usize,
+            &plaintext_bytes,
+        )
+        .map_err(|e| PyTypeError::new_err(format!("Error encrypting block: {e}")))?;
+
+    Ok(ciphertext)
+}
+
+/// Symmetrically Decrypt encrypted data in a block.
+#[pyfunction]
+pub fn decrypt_hybrid_block(
+    symmetric_key_bytes: Vec<u8>,
+    uid_bytes: Option<Vec<u8>>,
+    block_number: Option<usize>,
+    encrypted_bytes: Vec<u8>,
+) -> PyResult<Vec<u8>> {
+    //
+    // Check `user_decryption_key_bytes` input param and store it locally
+    if symmetric_key_bytes.len() != 32 {
+        return Err(PyTypeError::new_err("Symmetric key must be 32-bytes long"));
+    }
+
+    //
+    // Check `encrypted_bytes` input param and store it locally
+    if encrypted_bytes.is_empty() {
+        return Err(PyTypeError::new_err("Encrypted value is empty"));
+    }
+
+    //
+    // Parse symmetric key
+    let symmetric_key =
+        <Aes256GcmCrypto as SymmetricCrypto>::Key::try_from_bytes(symmetric_key_bytes)
+            .map_err(|e| PyTypeError::new_err(format!("Deserialize symmetric key failed: {e}")))?;
+
+    //
+    // Parse other input params
+    let uid = uid_bytes.map_or(vec![], |v| v.to_vec());
```

Comment:
```suggestion
    let uid = uid_bytes.unwrap_or(vec![]);
```

Reply:
`vec![]` or `Vec::new()` ? Maybe use the same for both occurrences (l.151 and l.196)

Reply:
I removed these options